openmdao/docs/features/building_blocks/components/external_code_comp.rst

```diff
@@ -61,6 +61,10 @@ does the same computation as the :ref:`Paraboloid Tutorial <tutorial_paraboloid_
 
 The following example demonstrates how to build an OpenMDAO component that makes use of this external code.
 
+.. note::
+    If you pass a string as a command, OpenMDAO sets `shell=True` which can add overhead leading to
+    a decrease in performance and a security loophole. Use list when possible.
```

Comment:
I would suggest linking to https://docs.python.org/3/library/subprocess.html#frequently-used-arguments, which contains this description:

```If shell is True, the specified command will be executed through the shell. This can be useful if you are using Python primarily for the enhanced control flow it offers over most system shells and still want convenient access to other shell features such as shell pipes, filename wildcards, environment variable expansion, and expansion of ~ to a user’s home directory. However, note that Python itself offers implementations of many shell-like features (in particular, glob, fnmatch, os.walk(), os.path.expandvars(), os.path.expanduser(), and shutil).```

as well as to https://docs.python.org/3/library/subprocess.html#security-considerations

(`shell=True` is not necessarily meaningful to the user without context)